### DISCUSSION TITLE
Have any best practices or conventions or suggestions for naming custom hooks?

### DISCUSSION THREAD

**Discussion by donniean** [TanStack/query#5198] [Q&A] [answered]
Have any best practices or conventions or suggestions for naming custom hooks?

For example, if I have 5 apis:

1. fetch device list
2. fetch device detail
3. create device
4. update device
5. delete device

How should I name my custom hooks?

`useDevices` or `useFetchDevices`
`useDevice` or `useFetchDevice` 
`useCreateDevice`
`useUpdateDevice`
`useDeleteDevice`

or

`useDevicesFetch`
`useDeviceFetch` 
`useDeviceCreate`
`useDeviceUpdate`
`useDeviceDelete`

or

`useDevicesQuery` or `useFetchDevicesQuery`
`useDeviceQuery` or `useFetchDeviceQuery` 
`useCreateDeviceMutation`
`useUpdateDeviceMutation`
`useDeleteDeviceMutation`

or

`useDevicesFetchQuery`
`useDeviceFetchQuery` 
`useDeviceCreateMutation`
`useDeviceUpdateMutation`
`useDeviceDeleteMutation`

or others?

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

I prefer the following:

`use(obviously)Operation(if applicable/necessary)Entity(Device, for example)Type(Query/Mutation)`

So, for example:

`useDeviceQuery`
`useDeleteDeviceMutation`

Having said that, there's no right or wrong. Use whatever you and your team prefer 🙂

**Comment by donniean** [comment]
anyone else ?
